src/store/actions/api.actions.js

```diff
@@ -93,27 +187,87 @@ export const deleteEndpoint = (apiName, callback) => async (dispatch) => {
         }));
         dispatch(callback(apiName));
     } catch (error) {
-        // console.log('ERROR => ', error);
         dispatch(openResponseModal({
             status: error.response.status,
             statusText: error.response.statusText,
             message: error.response.data.error,
         }));
-        // console.log('DELETE_ENDPOINT_ERROR', 'Infernal server error', error.response);
     }
 };
 
-export const fetchEndpoint = pathname => (dispatch) => {
+export const fetchEndpoint = pathname => async (dispatch) => {
     dispatch(getEndpointRequest());
 
-    return client.get(`apis${pathname}`)
-        .then((response) => {
-            dispatch(getEndpointSuccess(response.data));
-        })
-        .catch((error) => {
-            // eslint-disable-next-line
-            console.log('FETCH_ENDPOINT_ERROR', 'Infernal server error', error);
+    try {
+        const response = await client.get(`apis${pathname}`);
+        const preparedPlugins = response.data.plugins.map(plugin => {
+            if (plugin.name === 'rate_limit') {
```

Comment:
switch case here too

Reply:
Yeap, I'll do this in a separate **PR** in which I will also try to extract the common parts. This one is already too big :mask: